### ISSUE TITLE
Training doesn't work when using a generator

### ISSUE THREAD

**Issue by joksas** [joksas/nonideality-aware-mnn-training#4]
If I replace
https://github.com/awai54st/Non-Idealility-Aware-MBNN-Training/blob/f0c4036bab00efa3899deab87f8d87b78073bbc9/training/network.py#L40-L46
with
```python
if iterator.use_generator:
    if iterator.dataset=="CIFAR-10":
        horizontal_flip=True

    datagen = ImageDataGenerator(
            width_shift_range=0.15,  # randomly shift images horizontally (fraction of total width)
            height_shift_range=0.15,  # randomly shift images vertically (fraction of total height)
            horizontal_flip=horizontal_flip,  # randomly flip images
            validation_split=validation_split)
    training_data = datagen.flow(iterator.x_train, iterator.y_train, batch_size=iterator.training.batch_size, subset="training")
    validation_data = datagen.flow(iterator.x_train, iterator.y_train, batch_size=iterator.training.batch_size, subset="validation")
    history = model.fit(
            training_data,
            validation_data=validation_data,
            verbose=verbose,
            epochs=iterator.training.num_epochs,
            callbacks=[cback])
else:
    history=model.fit(
            iterator.x_train, iterator.y_train,
            batch_size=iterator.training.batch_size,
            validation_split=validation_split,
            verbose=verbose,
            epochs=iterator.training.num_epochs,
            callbacks=[cback])
```
the training accuracy stays at around 10% for CIFAR-10 (which utilises a generator).
